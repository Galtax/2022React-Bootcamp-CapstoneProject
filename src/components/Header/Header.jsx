import React from 'react';
import { Link } from 'react-router-dom'
import { ReactComponent as ShoppingCartIcon } from '../../images/shopping-cart.svg';
import logo from '../../assets/iconlogo.png';
import {
  HeaderMain,
  Image,
  Name,
  ButtonCart } from './Header.styled';
import { ROUTES } from '../../utils/constants';
import { InpurtSearch } from './InputSearch/InputSearch';

function Header() {

  return (
    <HeaderMain isYellow>
      <Link to={ROUTES.home}>
        <Image src={logo} alt="logo" />
      </Link>
      <Name>
        Galtax <span>  Luxury Furniture</span>
      </Name>
      <InpurtSearch />
      <ButtonCart>
        <ShoppingCartIcon />
      </ButtonCart>
    </HeaderMain>
  );
}

export default Header;
