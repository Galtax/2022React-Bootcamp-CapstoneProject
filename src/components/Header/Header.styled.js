import styled from 'styled-components';
import {
  bluePrince, white, 
} from '../../utils/constant.styles';

export const HeaderMain = styled.header`
  padding: 0 2rem;
  min-height: 65px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  position: fixed;
  background: #D4A373;
  top: 0;
  left: 0;
  right: 0;
  z-index: 31;

  @media (max-width: 576px) {
    justify-content: center;
  }
  }
  
  @media screen and (max-width: 800px){
    max-height: 20rem;
    flex-flow: row wrap;
    justify-content: space-between;
    padding-bottom: 2rem;
  }
`;

export const Name = styled.h1`
  font-size: 3.5rem;
  text-indent: 2rem;
  color: ${bluePrince};
  flex: 2 1 auto;
  font-weight: 600;
  height: 100%;
  display: flex;
  align-items: center;

  @media screen and (max-width: 800px) {
    justify-content: center;
    text-indent: 0;
    width: 65%;
  }
  @media screen and (max-width: 485px) {
    justify-content: center;
    text-indent: 0;
    width: fit-content;
  }
`;

export const ButtonCart = styled.button`
  margin: 0 4% 0;
  outline: none;
  background: none;
  border: none;
  path {
      fill: ${bluePrince};
    }
  &:hover {
    cursor: pointer;
    path {
      fill: ${white};
    }
  }

  & > svg {
    height: 2.5rem;
    width: 2.5rem;
  }

  @media screen and (max-width: 800px) {
    margin: 2rem auto 0;
  }
`;

export const Image = styled.img`
  display: flex;
  align-items: center;
  cursor: pointer;
  border-top-left-radius: .0rem;
  margin-left: 0;
  width: 10rem;
  height: 10rem;
  border-radius: 15px;

`;
