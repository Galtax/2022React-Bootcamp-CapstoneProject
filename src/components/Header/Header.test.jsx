import React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import Header from '.';
import { Name, Search, Image } from './Header.styled';

describe('Header Component Test', () => {

  it('should render the Header Component correctly', () => {
    const wrapper = mount(<Header />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain subcomponents', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find(Name).text()).toEqual('Furny');
    expect(wrapper.find(Image).exists()).toEqual(true);
    expect(wrapper.find(Search).exists()).toEqual(true);
  });
});
