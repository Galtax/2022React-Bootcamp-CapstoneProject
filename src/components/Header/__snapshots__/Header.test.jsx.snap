// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test should render the Header Component correctly 1`] = `ReactWrapper {}`;
